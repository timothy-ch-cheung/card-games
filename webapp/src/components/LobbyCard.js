import React from 'react';
import Card from "react-bootstrap/Card";
import Button from "react-bootstrap/Button";

function LobbyCard(props) {
    return (
        <div>
            <style type="text/css">
                {`
                    .lobby-card {
                        margin: 5px;
                    }
                `}
            </style>
            <Card className="lobby-card" style={{width: '18rem'}}>
                <Card.Body>
                    <Card.Title>{props.lobbyName}</Card.Title>
                    <Card.Text>
                        Host: {props.host}
                    </Card.Text>
                    <Button variant="info">Join Game</Button>
                </Card.Body>
            </Card>
        </div>
    );
}

export default LobbyCard;