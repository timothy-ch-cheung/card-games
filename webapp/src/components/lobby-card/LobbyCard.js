import React from 'react';
import Card from "react-bootstrap/Card";
import Button from "react-bootstrap/Button";
import {useSelector} from "react-redux";

function LobbyCard(props) {
    const userId = useSelector(state => state.user);

    const onJoinGame = () => {
        if (userId != null) {
            props.onSubmit(props.gameId, userId);
        } else {
            props.showModal(props.gameId);
        }
    }

    return (
        <>
            <style type="text/css">
                {`
                    .lobby-card {
                        margin: 5px;
                        width: 18rem;
                        flex-basis: 24%;
                    }
                    
                    @media(max-width: 1080px) {
                        .lobby-card {
                            flex-basis: 32%;
                        }
                    }
                    
                    @media(max-width: 800px) {
                        .lobby-card {
                            flex-basis: 48%;
                        }
                    }
                    
                    @media(max-width: 550px) {
                        .lobby-card {
                            flex-basis: 99%;
                        }
                    }
                    
                    .card-title {
                        border-bottom: solid 1px #d6d6d6;
                    }
                    
                `}
            </style>
            <Card className="lobby-card">
                <Card.Body>
                    <Card.Title className="card-title">{props.lobbyName}</Card.Title>
                    <Card.Text>
                        Host: {props.host}
                    </Card.Text>
                    <Button variant="info" onClick={onJoinGame}>Join Game</Button>
                </Card.Body>
            </Card>
        </>
    );
}

export default LobbyCard;