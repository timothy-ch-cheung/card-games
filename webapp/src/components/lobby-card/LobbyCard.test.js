import {configure, mount} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import LobbyCard from "../lobby-card/LobbyCard";
import React from "react";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";

configure({adapter: new Adapter()});

describe("TEST SUITE LobbyCard: ", () => {
    const mockStore = configureStore([]);
    let store;
    let submit;
    let showModal;

    beforeEach(() => {
        store = mockStore({
            user: null,
        });
        submit = jest.fn();
        showModal = jest.fn();
    })

    test('Matches LobbyCard snapshot', () => {
        const wrapper = mount(<Provider store={store}><LobbyCard/></Provider>);
        expect(wrapper).toMatchSnapshot()
    });

    test('join game opens create player modal when no player is stored', () => {
        const wrapper = mount(<Provider store={store}><LobbyCard onSubmit={submit} showModal={showModal}/></Provider>);
        wrapper.find('button[data-test="join-game-btn"]').simulate('click');
        expect(showModal).toHaveBeenCalledTimes(1);
        expect(submit).toHaveBeenCalledTimes(0);
    });

    test('join game opens lobby when player is stored', () => {
        store = mockStore({
            user: "12345678901234567890123456789012",
        });
        const wrapper = mount(<Provider store={store}><LobbyCard onSubmit={submit} showModal={showModal}/></Provider>);
        wrapper.find('button[data-test="join-game-btn"]').simulate('click');
        expect(submit).toHaveBeenCalledTimes(1);
        expect(showModal).toHaveBeenCalledTimes(0);
    });
});