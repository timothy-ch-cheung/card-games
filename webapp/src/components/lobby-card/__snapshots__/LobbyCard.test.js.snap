// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST SUITE LobbyCard:  Matches LobbyCard snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <LobbyCard>
    <style
      type="text/css"
    >
      
                    .lobby-card {
                        margin: 5px;
                        width: 18rem;
                        flex-basis: 24%;
                    }
                    
                    @media(max-width: 1080px) {
                        .lobby-card {
                            flex-basis: 32%;
                        }
                    }
                    
                    @media(max-width: 800px) {
                        .lobby-card {
                            flex-basis: 48%;
                        }
                    }
                    
                    @media(max-width: 550px) {
                        .lobby-card {
                            flex-basis: 99%;
                        }
                    }
                    
                    .card-title {
                        border-bottom: solid 1px #d6d6d6;
                    }
                    
                
    </style>
    <Card
      body={false}
      className="lobby-card"
    >
      <div
        className="lobby-card card"
      >
        <CardBody>
          <div
            className="card-body"
          >
            <CardTitle
              className="card-title"
            >
              <ForwardRef
                className="card-title card-title"
              >
                <div
                  className="card-title card-title h5"
                />
              </ForwardRef>
            </CardTitle>
            <CardText>
              <p
                className="card-text"
              >
                Host: 
              </p>
            </CardText>
            <Button
              active={false}
              data-test="join-game-btn"
              disabled={false}
              onClick={[Function]}
              type="button"
              variant="info"
            >
              <button
                className="btn btn-info"
                data-test="join-game-btn"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                Join Game
              </button>
            </Button>
          </div>
        </CardBody>
      </div>
    </Card>
  </LobbyCard>
</Provider>
`;
